bee-message/src/output/chain_id.rs

```diff
@@ -11,14 +11,41 @@ pub enum ChainId {
     ///
     Alias(AliasId),
     ///
+    Foundry(FoundryId),
+    ///
     Nft(NftId),
 }
 
+impl ChainId {
+    ///
+    pub fn is_null(&self) -> bool {
+        match self {
+            Self::Alias(alias_id) => alias_id.is_null(),
+            Self::Foundry(foundry_id) => foundry_id.is_null(),
+            Self::Nft(nft_id) => nft_id.is_null(),
+        }
+    }
+
+    ///
+    pub fn or_from_output_id(self, output_id: OutputId) -> ChainId {
+        if !self.is_null() {
+            return self;
+        }
+
+        match self {
+            Self::Alias(_) => Self::Alias(AliasId::from(output_id)),
+            Self::Foundry(_) => self,
+            Self::Nft(_) => Self::Nft(NftId::from(output_id)),
+        }
+    }
+}
+
 impl core::fmt::Display for ChainId {
     fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
         match self {
-            ChainId::Alias(alias) => write!(f, "{}", alias),
-            ChainId::Nft(nft) => write!(f, "{}", nft),
+            ChainId::Alias(id) => write!(f, "{}", id),
+            ChainId::Foundry(id) => write!(f, "{}", id),
+            ChainId::Nft(id) => write!(f, "{}", id),
```

Comment:
a little bit nit-picky, but reduces a bit of noise:
```rust
match self {
    ChaindId::Alias(id) => id,
    ....
}.fmt(f)
```

Reply:
All the match arms must return the same type so this is not legal Rust